src/tools/lib/python/picongpu/utils/find_time.py

```diff
@@ -0,0 +1,166 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017 PIConGPU contributors
+Authors: Axel Huebl
+License: GPLv3+
+"""
+
+import numpy as np
+import os
+
+
+class FindTime(object):
+    """
+    Convert iterations (time steps) to time [seconds] and back.
+    """
+
+    def __init__(self, run_directory):
+        """
+        Parameters
+        ----------
+        simulation_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        if run_directory is None:
+            raise ValueError('The run_directory parameter can not be None!')
+
+        self.run_directory = run_directory
+        self.data_file = "output"
+
+        self.dt = self.get_dt()
+
+    def get_data_path(self):
+        """
+        Return the path to the underlying data file.
+
+        Returns
+        -------
+        A string with a path.
+        """
+        sim_ouput_dir = os.path.join(self.run_directory, "simOutput")
+        if not os.path.isdir(sim_ouput_dir):
+            raise IOError('The simOutput/ directory does not exist inside '
+                          'path:\n  {}\n'
+                          'Did you set the proper path to the run directory?\n'
+                          'Did the simulation already run?'
+                          .format(self.run_directory))
+
+        data_file_path = os.path.join(sim_ouput_dir, self.data_file)
+        if not os.path.isfile(data_file_path):
+            raise IOError('The file {} does not exist.\n'
+                          'Did the simulation already run?'
+                          .format(data_file_path))
+
+        return data_file_path
```

Comment:
partly, I thought about this but since some might need an extra prefix for species and other might actually get a directory instead of a file I did not dig deeper yet.

But in principle yes, one can abstract that in the future into a common helper class.